### PR TITLE
Implement Group Class Scraper

### PR DESCRIPTION
**Pull request by kidzegeye**
## Overview

Implemented the scraper for group classes

## Changes Made

*Finished the scraper for group classes in `scraper.py`
* Created `ClassInstance` model, which is an Association table between `Classes` and `Gyms` that contains information about a scheduled instance of a class

## Test Coverage

Verified models and data in GraphQL playground and the Sqlite database

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  
  

</details>

### CODE REVIEW

**Review by joycrwu — COMMENTED** [review]
The scraper code looks good, for some of the models in the codebase, have their formatting been changed on purpose or is it just a git formatting thing?

**Comment by kidzegeye** [comment]
> The scraper code looks good, for some of the models in the codebase, have their formatting been changed on purpose or is it just a git formatting thing?

I think its just my linter

**Review by joycrwu — APPROVED** [review]
ok slay

**Review by isabellahoie — COMMENTED** [review]
Everything looks good just my 2 questions !!

File: src/models/classes.py [hunk 1]
```diff
@@ -50,3 +33,42 @@ def serialize(self):
...
+    id = Column(Integer, primary_key=True)
+    gym_id = Column(Integer, ForeignKey("gym.id"), nullable=True)
+    class_id = Column(Integer, ForeignKey("class.id"), nullable=False)
+    location = Column(String(), nullable=False)
+    instructor = Column(String(), nullable=False)
+    isCanceled = Column(Boolean(), nullable=False, default=False)
+    isVirtual = Column(Boolean(), nullable=False, default=False)
+    start_time = Column(DateTime(), nullable=False)
+    end_time = Column(DateTime(), nullable=False)
+    class_ = relationship("Class", back_populates="gyms")
```
[the remark this answers is not in the document]
Comment (kidzegeye) [reply src/models/classes.py:49]: its a relationship to the Class model, but "class" is a reservedword in python so i had to call it class_

File: src/models/classes.py [hunk 2]
```diff
@@ -50,3 +33,42 @@ def serialize(self):
...
+    gym = relationship("Gym", back_populates="classes")
```
[the remark this answers is not in the document]
Comment (kidzegeye) [reply src/models/classes.py:50]: many classInstances to one gym

[the remark this answers is not in the document]
Comment (isabellahoie) [reply src/models/classes.py:50]: wait so what is the difference between the "gym_id" column and "gym" column

**Review by isabellahoie — APPROVED** [review]
all good :)

**Pull request merged**